src/modules/sdbus/message.h

```diff
@@ -0,0 +1,29 @@
+/************************************************************\
+ * Copyright 2023 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _SDBUS_MESSAGE_H
+#define _SDBUS_MESSAGE_H
+
+#include <systemd/sd-bus.h>
+#include <jansson.h>
+
+const char *sdmsg_typestr (sd_bus_message *m);
+
+int sdmsg_put (sd_bus_message *m, const char *fmt, json_t *o);
+
+int sdmsg_write (sd_bus_message *m, const char *fmt, json_t *o);
+
+int sdmsg_get (sd_bus_message *m, const char *fmt, json_t **op);
+
+int sdmsg_read (sd_bus_message *m, const char *fmt, json_t *o);
```

Comment:
perhaps add comments on return values, since -errno is not the most common expected return.  and `sdmsg_read()` returns 1 on success instead of 0?  Perhaps follows another function from `sd_bus`?

Reply:
Great point.  I was following libsystemd's error conventions so these functions can be easily mixed with `sd_bus_*` ones.  I should have some comment to that effect inline.